test-cli.py

```diff
@@ -742,65 +769,93 @@ def stress(use_time, request_time, use_memory, request_memory, use_disk, request
-    run_command([cl, 'add', 'bundle', uuid, remote_worksheet])
-    check_agree([cl, 'info', '-f', 'data_hash,data_size,name', uuid])
-    check_agree([cl, 'cat', uuid])
-
-    # Upload to remote, transfer to local
-    run_command([cl, 'work', remote_worksheet])
-    uuid = run_command([cl, 'upload', test_path('')])
-    run_command([cl, 'add', 'bundle', uuid, local_worksheet])
-    check_agree([cl, 'info', '-f', 'data_hash,data_size,name', uuid])
-    check_agree([cl, 'cat', uuid])
-
-    # Upload to remote, transfer to local (metadata only)
-    run_command([cl, 'work', remote_worksheet])
-    uuid = run_command([cl, 'upload', '-c', 'hello'])
-    run_command([cl, 'rm', '-d', uuid])  # Keep only metadata
-    run_command([cl, 'add', 'bundle', uuid, local_worksheet])
-
-    # Upload to local, transfer to remote (metadata only)
-    run_command([cl, 'work', local_worksheet])
-    uuid = run_command([cl, 'upload', '-c', 'hello'])
-    run_command([cl, 'rm', '-d', uuid])  # Keep only metadata
-    run_command([cl, 'add', 'bundle', uuid, remote_worksheet])
-
-    # Test adding worksheet items
-    run_command([cl, 'wadd', local_worksheet, remote_worksheet])
-    run_command([cl, 'wadd', remote_worksheet, local_worksheet])
-
-    # Cleanup
-    del os.environ['CODALAB_HOME']
-    run_command([cl, 'work', remote_worksheet])
-    shutil.rmtree(home)
+    remote_home = temp_path('-home')
+    os.environ['CODALAB_HOME'] = remote_home
+    os.mkdir(remote_home)
+
+    test_username = os.environ.get('CODALAB_USERNAME', 'codalab')
+    test_password = os.environ.get('CODALAB_PASSWORD', '1234')
+
+    # Initialize new config
+    subprocess.call('printf "n\nn\n" | scripts/create-root-user.py ' + test_password, shell=True)
+    with edit_json(os.path.join(remote_home, 'config.json')) as config:
+        config['server']['port'] = 3800
+        config['server']['rest_port'] = 3900
+    remote_worksheet = 'http://localhost:3800::'
+
+    # Start auxiliary servers for the new auxiliary host
+    os.environ['PYTHONUNBUFFERED'] = '1'  # prevents stalling when waiting for output
+    server_proc = subprocess.Popen([cl, 'server'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    rest_server_proc = subprocess.Popen([cl, 'rest-server'], stderr=subprocess.PIPE)
+
+    # TODO(sckoo): refactor test-cli to allow each module to define cleanup method
+    try:
+        wait_until_substring(server_proc.stdout, 'BundleRPCServer serving')
+        wait_until_substring(rest_server_proc.stderr, 'Booting worker')
+
+        # Restore original config
+        os.environ['CODALAB_HOME'] = old_home
+
+        # Switch to new host and log in to cache auth token
+        subprocess.call([cl, 'logout', remote_worksheet[:-2]])
+        os.environ['CODALAB_USERNAME'] = test_username
+        os.environ['CODALAB_PASSWORD'] = test_password
+        subprocess.call('cl work ' + remote_worksheet, shell=True)
+        subprocess.call('cl gls > /dev/null', shell=True)
```

Comment:
Just the first rest-ified command I could think of. I'll give a shot at fixing the double-login thing now, but if it doesn't go well I'll make sure to document this line better.